lib/daru/io/exporters/json.rb

```diff
@@ -0,0 +1,403 @@
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "common": {
+        #   #       "gender": "Male",
+        #   #       "age": 54
+        #   #     },
+        #   #     "specific": {
+        #   #       "name": "Rhaegar Targaryen"
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "common": {
+        #   #       "gender": "Female",
+        #   #       "age": 36
+        #   #     },
+        #   #    "specific": {
+        #   #       "name": "Lyanna Stark"
+        #   #     }
+        #   #   }
+        #   # ]
+        # @example Writing to a JSON file with dynamic JsonPaths
+        #   df = Daru::DataFrame.new(
+        #     [
+        #       {name: 'Jon Snow', age: 18, sex: 'Male'},
+        #       {name: 'Rhaegar Targaryen', age: 54, sex: 'Male'},
+        #       {name: 'Lyanna Stark', age: 36, sex: 'Female'}
+        #     ],
+        #     order: %i[name age sex],
+        #     index: %i[child dad mom]
+        #   )
+        #
+        #   #=> #<Daru::DataFrame(3x3)>
+        #   #            name          age       sex
+        #   # child   Jon Snow         18       Male
+        #   # dad   Rhaegar Ta         54       Male
+        #   # mom   Lyanna Sta         36     Female
+        #
+        #   Daru::IO::Exporters::JSON.new(
+        #     df,
+        #     'filename.json',
+        #     orient: :records,
+        #     pretty: true,
+        #     age: '$..{name}..age',
+        #     sex: '$..{name}..gender'
+        #   )
+        #
+        #   #=>
+        #   # [
+        #   #   {
+        #   #     "Jon Snow": {
+        #   #       "gender": "Male",
+        #   #       "age": 18
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "Rhaegar Targaryen": {
+        #   #       "gender": "Male",
+        #   #       "age": 54
+        #   #     }
+        #   #   },
+        #   #   {
+        #   #     "Lyanna Stark": {
+        #   #     "gender": "Female",
+        #   #     "age": 36
+        #   #     }
+        #   #   }
+        #   # ]
+        def initialize(dataframe, path, orient: :index, pretty: false, **jsonpaths)
+          optional_gem 'json'
+          optional_gem 'jsonpath'
+
+          super(dataframe)
+          @path          = path
+          @orient        = orient
+          @pretty        = pretty
+          @jsonpath_hash = jsonpaths.empty? ? nil : jsonpaths
+        end
+
+        def call
+          @jsonpath_hash ||= @dataframe.vectors.to_a.map { |v| {v => "$..#{v}"} }.reduce(:merge)
+          @vectors         = @jsonpath_hash.keys
+          @jsonpaths       = process_jsonpath
+          @json_content    = process_json_content
+
+          File.open(@path, 'w') { |file| file.write(process_json_string) }
+        end
+
+        private
+
+        def process_json_string
+          return ::JSON.pretty_generate(@json_content) if @pretty
+          ::JSON.generate(@json_content)
```

Comment:
As I've said in another PR, I believe you are too inclined to the pattern sometimes. For me, this version is cleaner:
```ruby
@pretty ? ::JSON.pretty_generate : ::JSON.generate
```
If you really want to, you can "hack" through Ruby code and even do something like:
```ruby
::JSON.generate(@json_content, @pretty ? JSON::PRETTY_STATE_PROTOTYPE : {})
```
...but, honestly, I am not sure it is better :)